src/main/scala/com/fulcrumgenomics/bam/IdentifyPrimers.scala

```diff
@@ -0,0 +1,1027 @@
+    |location of their respective primers.
+    |2. Assign to a non-canonical primer pair if the 5' end of the mates agree with the mapping location of their
+    |respective primers, and the primer and read sequences "match" respectively for each primer.
+    |3. Assign each mate of a read pair to a primer based on fully-gapped alignment.
+    |
+    |## Paired-End Matching (5' and 3')
+    |
+    |The `--three-prime` option can be used to also search the 3' end of every read for a primer sequence as follows:
+    |
+    |1. If a primer pair is identified using the 5' end, that pair is searched on the 3' end.
+    |2. Otherwise, assign a primer based on fully-gapped alignment.
+    |
+    |The primer matched on the 5' end of read one must match the primer matched on the 3' end of read two, and the
+    |primer matched on the 3' end of read one must match the primer matched on the 5' end of read two.  This will only
+    |be enforced if both ends of a read match a primer.
+    |
+    |## Fragment Matching (5' only)
+    |
+    |If the input are fragment reads, then only the 5' end of a read is used to match against the input list of primers:
+    |
+    |1. Assign to a primer if the 5' end agrees with the mapping location of the respective primer, and the primer
+    |and read sequence "match".
+    |2. Assign to a primer based on fully-gapped alignment.
+    |
+    |## Fragment Matching (5' and 3')
+    |
+    |The 3' end of the read can also be searched to identify primer pairs using the `--three-prime` option.  In this
+    |case the same algorithm will be employed as paired-end matching for the 5' only case, where each end of the fragment
+    |read is equivalent to the 5' end of the mates of the paired end rad.
+    |
+    |## Unmapped Data
+    |
+    |If no reference is given or mapping information is not available, matching using mapping locations is skipped.
+    |
+    |## Speeding Up Alignment
+    |
+    |Install [the ksw executable](https://github.com/nh13/ksw) manually or via conda (`conda install -c bioconda ksw`)
+    |and supply the path to the `ksw` executable via `--ksw`.
+  """)
+class IdentifyPrimers
+(@arg(flag='i', doc="Input BAM file.")  val input: PathToBam,
+ @arg(flag='p', doc="File containing information about the primers.") val primers: FilePath,
+ @arg(flag='o', doc="Output BAM file.") val output: PathToBam,
+ @arg(flag='m', doc="Path prefix for the metrics files.") val metrics: PathPrefix,
+ @arg(flag='S', doc="Match to primer locations +/- this many bases.") val slop: Int = 5,
+ @arg(          doc="Search for primers on the 3' end of each read as well (5' only by default).") val threePrime: Boolean = false,
+ @arg(          doc="Maximum mismatches for a primer to be considered a match.") val maxMismatches: Int = 1,
+ @arg(          doc="The minimum alignment score for fully gapped alignment for a primer to be considered a match.") val minAlignmentScore: Int = 10,
```

Comment:
I'm not convinced that users will be good at supplying a cutoff alignment score.  Though I guess if you don't get the mismatches and indels, there's not much else you can do.